plugin/src/main/java/org/opensearch/ml/rest/RestMLSearchConnectorAction.java

```diff
@@ -10,15 +10,22 @@
 
 import org.opensearch.ml.common.connector.Connector;
 import org.opensearch.ml.common.transport.connector.MLConnectorSearchAction;
+import org.opensearch.ml.settings.MLFeatureEnabledSetting;
 
 import com.google.common.collect.ImmutableList;
 
 public class RestMLSearchConnectorAction extends AbstractMLSearchAction<Connector> {
     private static final String ML_SEARCH_CONNECTOR_ACTION = "ml_search_connector_action";
     private static final String SEARCH_CONNECTOR_PATH = ML_BASE_URI + "/connectors/_search";
 
-    public RestMLSearchConnectorAction() {
-        super(ImmutableList.of(SEARCH_CONNECTOR_PATH), ML_CONNECTOR_INDEX, Connector.class, MLConnectorSearchAction.INSTANCE);
+    public RestMLSearchConnectorAction(MLFeatureEnabledSetting mlFeatureEnabledSetting) {
+        super(
+            ImmutableList.of(SEARCH_CONNECTOR_PATH),
```

Comment:
(Observation) Java has had `List.of()` since JDK9.  I wonder if you can get rid of the whole Guava dependency with a global search and replace.